tests/unit/package_managers/test_generic.py

```diff
@@ -39,10 +75,79 @@ def test_fetch_generic_source(
     mock_from_obj_list.assert_called_with(components=components)
 
 
-def test_resolve_generic_no_lockfile(rooted_tmp_path: RootedPath) -> None:
+@mock.patch("cachi2.core.package_managers.generic.main._load_lockfile")
+def test_resolve_generic_no_lockfile(mock_load: mock.Mock, rooted_tmp_path: RootedPath) -> None:
     with pytest.raises(PackageRejected) as exc_info:
         _resolve_generic_lockfile(rooted_tmp_path, rooted_tmp_path)
     assert (
         f"Cachi2 generic lockfile '{DEFAULT_LOCKFILE_NAME}' missing, refusing to continue"
         in str(exc_info.value)
     )
+    mock_load.assert_not_called()
+
+
+@pytest.mark.parametrize(
+    ["lockfile", "expected_exception", "expected_err"],
+    [
+        pytest.param("{", PackageRejected, "yaml format is not correct", id="invalid_yaml"),
+        pytest.param(
+            LOCKFILE_WRONG_VERSION, PackageManagerError, "Input should be '1.0'", id="wrong_version"
+        ),
+        pytest.param(
+            LOCKFILE_CHECKSUM_MISSING, PackageManagerError, "Field required", id="no_checksum_1"
+        ),
+        pytest.param(
+            LOCKFILE_CHECKSUM_EMPTY,
+            PackageManagerError,
+            "At least one checksum must be provided",
+            id="no_checksum_2",
+        ),
+    ],
+)
+def test_resolve_generic_lockfile_invalid(
+    lockfile: str,
+    expected_exception: Type[Cachi2Error],
+    expected_err: str,
+    rooted_tmp_path: RootedPath,
+) -> None:
+    # setup lockfile
+    with open(rooted_tmp_path.join_within_root(DEFAULT_LOCKFILE_NAME), "w") as f:
+        f.write(lockfile)
+
+    with pytest.raises(expected_exception) as exc_info:
+        _resolve_generic_lockfile(rooted_tmp_path, rooted_tmp_path)
+
+    assert expected_err in str(exc_info.value)
+
+
+@pytest.mark.parametrize(
+    ["lockfile", "expected_lockfile"],
+    [
+        pytest.param(
+            LOCKFILE_VALID,
+            GenericLockfileV1.model_validate(
+                {
+                    "metadata": {"version": "1.0"},
+                    "artifacts": [
+                        {
+                            "download_url": "https://redhat.com/artifact",
```

Comment:
Don't need parametrize for a single test case.

Reply:
I know I don't need to, but this feels cleaner, since it separates the data from the test logic. I've received the opposite instructions on the previous [PR](https://github.com/containerbuildsystem/cachi2/pull/671#discussion_r1787518942). If this is a blocker I don't mind changing it.